### PR TITLE
Feature/172/categories title with description and search input

### LINKED ISSUE #172: (SP: 1) [Categories] Add title with description and search input
UserStory: #52 
## Description
Add title with description, implement input with searching functionality to search for a category by name
## Design 
image
## Acceptance Criteria
- [ ] When a user clicks on the "Show all offers", he/she should be redirected to "find offers" page
- [ ] When a user clicks on the "Search" button or press enter, search should be performed

### PR DESCRIPTION
**Pull request by OlhaOvsiienko**
image

### CODE REVIEW

**Comment by YaroslavChuiko** [comment]
image

what is that black line?

File: src/containers/categories-title-input/CategoriesTitleInput.jsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            startAdornment={
+              <InputAdornment position='start'>
+                <SearchIcon />
+              </InputAdornment>
+            }
+            sx={styles.inputField}
+            type='text'
+            value={categoryName}
+            variant='outlined'
+          />
```
Comment (YaroslavChuiko) [inline src/containers/categories-title-input/CategoriesTitleInput.jsx]: Have you tried to use the AppTextField component?

File: src/containers/categories-title-input/CategoriesTitleInput.jsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+const CategoriesTitleInput = () => {
+  const { t } = useTranslation()
+  const navigate = useNavigate()
+  const [categoryName, setCategoryName] = useState('')
+
+  const { response, error, loading, fetchData } = useAxios({
+    service: categoryService.getCategories,
+    defaultResponse: null,
+    fetchOnMount: false
+  })
```
Comment (YaroslavChuiko) [inline src/containers/categories-title-input/CategoriesTitleInput.jsx]: You don't need to fetch and show the category list in this component, just set categoryName as the search param, and the CategoryList component will get this value from there and fetch data

File: src/containers/categories-title-input/CategoriesTitleInput.jsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+                  title={item.name}
+                  totalOffers={
+                    item.totalOffers.student + item.totalOffers.tutor
+                  }
+                />
+              )
+            })}
+          </Box>
+        )}
+      </Box>
```
Comment (YaroslavChuiko) [inline src/containers/categories-title-input/CategoriesTitleInput.jsx]: results of searching will be shown in the CategoryList component

**Review by YaroslavChuiko — CHANGES_REQUESTED** [review]

File: src/containers/categories-title-input/CategoriesTitleInput.jsx (hunk 1 above)
Comment (OlhaOvsiienko) [reply src/containers/categories-title-input/CategoriesTitleInput.jsx]: I used AppTextField component in the update version of the CategoryTitleInput. Thank you!

File: src/containers/categories-title-input/CategoriesTitleInput.jsx (hunk 2 above)
Comment (OlhaOvsiienko) [reply src/containers/categories-title-input/CategoriesTitleInput.jsx]: I did it. Thanks!

File: src/containers/categories-title-input/CategoriesTitleInput.jsx (hunk 3 above)
Comment (OlhaOvsiienko) [reply src/containers/categories-title-input/CategoriesTitleInput.jsx]: This part has been removed

**Review by YaroslavChuiko — APPROVED** [review]

**Review by khytrykv — APPROVED** [review]

File: src/containers/categories-title-input/CategoriesTitleInput.jsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+import { Box, InputAdornment, Typography } from '@mui/material'
+import TitleWithDescription from '~/components/title-with-description/TitleWithDescription'
+import AppButton from '~/components/app-button/AppButton'
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward'
+import SearchIcon from '@mui/icons-material/Search'
+import { useTranslation } from 'react-i18next'
+import { useNavigate } from 'react-router-dom'
+import { authRoutes } from '~/router/constants/authRoutes'
+import { styles } from '~/containers/categories-title-input/CategoriesTitleInput.styles'
+import { useSearchParams } from 'react-router-dom'
```
Comment (vlad-khrychov) [inline src/containers/categories-title-input/CategoriesTitleInput.jsx]: On 8th line we have import from 'react-router-dom', so combine this two imports. 
```suggestion
import { useNavigate, useSearchParams } from 'react-router-dom'
```

**Review by vlad-khrychov — CHANGES_REQUESTED** [review]

File: src/containers/categories-title-input/CategoriesTitleInput.jsx (hunk 4 above)
Comment (OlhaOvsiienko) [reply src/containers/categories-title-input/CategoriesTitleInput.jsx]: I combined these two imports.
Thank you!

**Review by YaroslavChuiko — APPROVED** [review]

**Review by khytrykv — APPROVED** [review]

**Review by vlad-khrychov — APPROVED** [review]

**Pull request merged**